### PR TITLE
Switch doxygen documentation theme

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Use modern [doxygen-awesome-css](https://jothepro.github.io/doxygen-awesome-css/index.html#autotoc_md10) theme (as mentioned in https://github.com/Cantera/enhancements/issues/115#issuecomment-1477177638) and put details on top via [customized xml layout](https://stackoverflow.com/questions/19831837/documenting-with-doxygen-put-details-on-top)

**Rendered Webpage**

image

vs. current theme (after #1543)

image

**Relevant Issues**

Somewhat relevant to Cantera/enhancements#115

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
I added a couple of tweaks that became possible with `DoxygenLayout.xml`, and fixed a couple of formatting glitches I stumbled across. I am not planning on adding anything else, i.e. this is ready for a review. (I also checked that the GH doxygen artifacts look as expected).

File: doc/doxygen/Doxyfile [hunk 1]
```diff
@@ -107,7 +107,7 @@ BRIEF_MEMBER_DESC      = YES
 # brief descriptions will be completely suppressed.
 # The default value is: YES.
 
-REPEAT_BRIEF           = YES
+REPEAT_BRIEF           = NO
```
Comment (speth) [inline doc/doxygen/Doxyfile]: I think changing this setting makes the detailed member documentation much worse. For example:

image

**Review by speth — CHANGES_REQUESTED** [review]
Thanks, @ischoegl! This is definitely an improvement over the default styling. While I suspect there are some additional tweaks we may want to make in time, there's just one issue that I think should make now, noted below.

File: doc/doxygen/Doxyfile (hunk 1 above)
Comment (ischoegl) [reply doc/doxygen/Doxyfile]: Good catch! A lot of this can be fine-tuned by customizing the XML file.

**Review by speth — APPROVED** [review]

**Pull request merged**